tests/acceptance/features/bootstrap/GraphContext.php

```diff
@@ -1532,4 +1532,90 @@ public function theResponseShouldContainTheFollowingDriveInformation(TableNode $
 			throw new Error('Response is not an array or the array does not consist key "drive"');
 		}
 	}
+
+	/**
+	 * @When user :user gets details of a group :groupName using the Graph API
+	 *
+	 * @param string $user
+	 * @param string $groupName
+	 *
+	 * @return void
+	 */
+	public function userGetsDetailsOfAGroupUsingTheGraphApi(string $user, string $groupName): void {
+		$groupId = $this->getGroupIdByName($this->getArrayOfGroupsResponded($this->listGroups($user)), $groupName);
+		$credentials = $this->getAdminOrUserCredentials($user);
+
+		$this->featureContext->setResponse(
+			GraphHelper::getSingleGroup(
+				$this->featureContext->getBaseUrl(),
+				$groupId,
+				$this->featureContext->getStepLineRef(),
+				$credentials["username"],
+				$credentials["password"]
+			)
+		);
+	}
+
+	/**
+	 * Get Group Id from response
+	 *
+	 * @param array|null $allGroups
+	 * @param string $groupName - name of the group
+	 *
+	 * @return string - id of the group
+	 * @throws Exception | GuzzleException
+	 */
+	public function getGroupIdByName(?array $allGroups, string $groupName): string {
+		if ($allGroups === null) {
+			$allGroups = $this->getArrayOfGroupsResponded(
+				$this->listGroups(
+					$this->featureContext->getCurrentUser()
+				)
+			);
+		}
+		foreach ($allGroups as $group) {
+			if ($group['displayName'] === $groupName) {
+				return $group['id'];
+			}
+		}
+		throw new Exception("Group with name $groupName not found in JSON response.");
+	}
+
+	/**
+	 * @Then the json responded should contain a group with these key and value pairs:
+	 *
+	 * @param TableNode $table
+	 *
+	 * @return void
+	 */
+	public function theJsonRespondedShouldContainAGroupWithTheseKeyAndValuePairs(TableNode $table): void {
+		$actualGroup = $this->featureContext->getJsonDecodedResponse();
+		$this->featureContext->verifyTableNodeColumns($table, ['key', 'value']);
+
+		$expectedGroup = [];
+		foreach ($table->getHash() as $row) {
```

Comment:
`getRowsHash` suitable for this type of data table
```suggestion
		foreach ($table->getRowsHash() as $row) {
```